components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/LocalEntryResource.java

```diff
@@ -67,15 +75,50 @@ public boolean invoke(MessageContext messageContext,
                           SynapseConfiguration synapseConfiguration) {
 
         String entryKey = Utils.getQueryParameter(messageContext, Constants.NAME);
-        if (Objects.nonNull(entryKey)) {
-            populateLocalEntryData(axis2MessageContext, synapseConfiguration, entryKey);
-        } else {
-            populateLocalEntries(axis2MessageContext, synapseConfiguration);
+        String searchKey = Utils.getQueryParameter(messageContext, SEARCH_KEY);
+
+        try {
+            if (Objects.nonNull(entryKey)) {
+                populateLocalEntryData(axis2MessageContext, synapseConfiguration, entryKey);
+            } else if (Objects.nonNull(searchKey) && !searchKey.trim().isEmpty()) {
+                populateSearchResults(messageContext, searchKey.toLowerCase());
+            } else {
+                populateLocalEntries(axis2MessageContext, synapseConfiguration);
+            }
+        } catch (AxisFault e) {
+            LOG.error("Error while populating service: ", e);
```

Comment:
We are returning true when there is an exception. Is that intentional? Are we checking this boolean in the calling method?